### PR TITLE
Add missing param declarations (Ubuntu, Windows)

### PR DESCRIPTION
**Pull request by bittner**
Tests on the roles of our control repo discovered that some param values are not set for Ubuntu and Windows. This PR adds them.

### CODE REVIEW

**Comment by bittner** [comment]
Interestingly, the tests fail even with the branch containing the fixes: :confused: 

```ruby
$ pdk test unit
[✔] Preparing to run the unit tests.
[✖] Running unit tests.
  Evaluated 4 tests in 3.484193657 seconds: 2 failures, 0 pending.
[✔] Cleaning up after running unit tests.
failed: rspec: ./spec/classes/role_workstation_backoffice_spec.rb:11: error during compilation: Evaluation Error: Unknown variable: 'software::params::firefox_version'. at /home/peter.bittner/Development/repos/puppet-control/spec/fixtures/modules/software/manifests/browsers/firefox.pp:8:14 on node peter-pc.comsolit.local
  role::workstation::backoffice on ubuntu-16.04-x86_64 with defaults should compile into a catalogue without dependency cycles
  Failure/Error:
  
        context 'with defaults' do
          it { is_expected.to compile.with_all_deps }
        end
      end

failed: rspec: ./spec/classes/role_workstation_developer_spec.rb:11: error during compilation: Evaluation Error: Unknown variable: 'software::params::firefox_version'. at /home/peter.bittner/Development/repos/puppet-control/spec/fixtures/modules/software/manifests/browsers/firefox.pp:8:14 on node peter-pc.comsolit.local
  role::workstation::developer on ubuntu-16.04-x86_64 with defaults should compile into a catalogue without dependency cycles
  Failure/Error:
  
        context 'with defaults' do
          it { is_expected.to compile.with_all_deps }
        end
      end
```
I have a class `profile::workstation` that both roles include, and it starts like this:

```puppet
class profile::workstation {
  include software::browsers::chrome
  include software::browsers::firefox
```

Any idea what could be wrong, @edestecd?

**Comment by bittner** [comment]
Don't worry about my comment, got this fixed. The problem was in `.fixtures.yml` where I used `ref:` indead of `branch:`, e.g.
```yaml
fixtures:
  forge_modules:
    software:
      repo: 'https://github.com/comsolit/puppet-software'
      branch: 'fix/add-missing-undefs'
```
I'll delete the comment above to reduce confusion.

**Comment by bittner** [comment]
Note that this issue is related to [STRICT_VARIABLES](https://github.com/puppetlabs/puppetlabs_spec_helper#initializing-puppet-for-testing) in the spec helper, which _could_ be turned on as an alternative. A very similar scenario is discussed in https://github.com/joshuabaird/puppet-ipaclient/issues/63 (custom facts).

**Comment by bittner** [comment]
Puppet Lab's PDK has changed opinion about supported OSes and makes test runs fail with the latest version. The second commit addresses this issue and makes Debian a supported option in general.

See also: https://github.com/puppetlabs/pdk/issues/338

**Comment by bittner** [comment]
@edestecd This PR is ready for merging. Thank you!

**Comment by bittner** [comment]
Anything I still need to do to make you merge this PR?

File: .fixtures.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 fixtures:
-  repositories:
-    stdlib: "https://github.com/puppetlabs/puppetlabs-stdlib.git"
-    apt: "https://github.com/puppetlabs/puppetlabs-apt.git"
-    vcsrepo: "https://github.com/puppetlabs/puppetlabs-vcsrepo.git"
```
Comment (edestecd) [inline .fixtures.yml]: hmm why are we switching these? It seems like puppetlabs modules still use the repos.
https://github.com/puppetlabs/puppetlabs-apache/blob/master/.fixtures.yml

I also like testing against newer code, so we can be aware sooner of upcoming changes.

File: manifests/editors/atom.pp [hunk 2]
```diff
@@ -52,6 +52,23 @@
...
     }
+    'Debian': {
+      validate_string($url)
+      $apm_require = Package['atom']
+
+      archive { 'atom-amd64.deb':
+        source  => 'https://atom.io/download/deb',
+        path    => '/tmp/atom-amd64.deb',
+        extract => false,
+        cleanup => false,
```
Comment (edestecd) [inline manifests/editors/atom.pp]: Hmm I need to investigate this more.  I have seen examples of downloading debs before...
Not so sure we need to keep downloading it to /tmp once it is installed. I think some other solutions I have seen solve that somehow. I need to go digging

Comment (edestecd) [reply manifests/editors/atom.pp]: Also in puppet 4 the file type can take http urls as source now. If there is no extraction, we may want to use that to reduce dependencies

**Comment by edestecd** [comment]
@bittner sorry for the delay. I have a few comments.

File: .fixtures.yml (hunk 1 above)
Comment (bittner) [reply .fixtures.yml]: Oh, that's true! It's for testing, then it makes sense. (I thought I would do you a favor, sorry!) :flushed:

**Comment by bittner** [comment]
I hope the changes now do what they should. Thanks for your input!

File: manifests/editors/atom.pp [hunk 3]
```diff
@@ -52,6 +52,24 @@
...
+      $apm_require = Package['atom']
+
+      package { 'atom':
+        ensure   => $ensure,
+        source   => '/tmp/atom-amd64.deb',
+        provider => 'dpkg',
+      }
+
+      file { '/tmp/atom-amd64.deb':
+        source => 'https://atom.io/download/deb',
```
Comment (bittner) [inline manifests/editors/atom.pp]: On Puppet 4.8 this doesn't work for some reason. Despite the [Puppet documentation](https://puppet.com/docs/puppet/4.8/types/file.html#file-attribute-source) saying it should work. I had to fall back to using `archive` for downloading the .deb file. Not sure why. I remember trying hard before giving up.

Comment (bittner) [reply manifests/editors/atom.pp]: But then again: We're using _this_ module, _this_ branch in production. And it _works_. :smirk: :astonished: 

I think the world is just flat, really.

**Comment by bittner** [comment]
Do you have any more comments before you can merge this PR?

**Comment by bittner** [comment]
@edestecd We want to open more PRs, and the changes in this PR would be needed to continue without fuss. Would you mind to merge it?

**Comment by bittner** [comment]
This PR now also updates the Skype installation. Now the `skypeforlinux` package is installed from official Microsoft repositories for both vanilla Debian and Ubuntu.

File: manifests/editors/atom.pp (hunk 3 above)
Comment (edestecd) [reply manifests/editors/atom.pp]: lol, it works for us in our environment.

File: metadata.json [hunk 4]
```diff
@@ -17,6 +17,7 @@
   "dependencies": [
     {"name": "puppetlabs/stdlib", "version_requirement": ">= 4.2.0 < 5.0.0"},
     {"name": "puppetlabs/apt", "version_requirement": ">= 2.1.1 < 3.0.0"},
+    {"name": "puppet/archive", "version_requirement": ">= 1.1.2 < 3.0.0"},
```
Comment (edestecd) [inline metadata.json]: Are we using archive?

Comment (edestecd) [reply metadata.json]: I can merge if you remove this.

Comment (bittner) [reply metadata.json]: Now we have a few new errors from existing code. This would go away anyway if you merge #15, I would guess.

**Comment by edestecd** [comment]
The new errors are from a new rubocop check that was added today.
We hit it in our puppet control repo today as well.

**Comment by bittner** [comment]
Thanks!

I'm now rebasing PR #15. Can you take a look at that later, too?

**Pull request merged**